File: src/exercises/clock/typescript/template.ts
export class Clock {
  constructor(hour: number, minute: number = 0) {
    // Your code here
  }

  public toString(): string {
    // Your code here
    return "00:00";
  }

  public plus(minutes: number): Clock {
    // Your code here
    return this;
  }

  public minus(minutes: number): Clock {
    // Your code here
    return this;
  }

  public equals(other: Clock): boolean {
    // Your code here
    return false;
  }
}

File: src/exercises/clock/python/template.py
class Clock:
    def __init__(self, hour: int, minute: int = 0):
        # Your code here
        pass

    def __repr__(self) -> str:
        return "00:00"

    def __eq__(self, other: object) -> bool:
        return False

    def __add__(self, minutes: int) -> "Clock":
        return self

    def __sub__(self, minutes: int) -> "Clock":
        return self
